src/syscalls/mod.rs

```diff
@@ -0,0 +1,99 @@
+#[cfg_attr(target_arch = "riscv32", path = "syscalls_riscv32.rs")]
+#[cfg_attr(target_arch = "arm", path = "syscalls_arm.rs")]
+#[cfg_attr(
+    not(any(target_arch = "arm", target_arch = "riscv32")),
+    path = "syscalls_mock.rs"
+)]
+pub mod raw;
+
+use crate::callback::CallbackSubscription;
+use crate::callback::SubscribableCallback;
+use crate::shared_memory::SharedMemory;
+
+/// # Safety
+///
+/// Yielding in the main function should be safe. Nevertheless, yielding manually is not required as this is already achieved by the `async` runtime.
+///
+/// When yielding in callbacks, two problems can arise:
+/// - The guarantees of `FnMut` are violated. In this case, make sure your callback has `Fn` behavior.
+/// - Callbacks can get executed in a nested manner and overflow the stack quickly.
+#[export_name = "libtock::syscalls::yieldk"]
```

Comment:
Why is export_name here?

Reply:
Right. Makes no sense.

Reply:
Is your question "Why is there an export_name at all?" or "Why is it here and not somewhere else?" ?

Reply:
Yes, why does it exist. It looks like you're doing some sort of manual linking with the `async_support` crate?

Reply:
Yes, it's a back-channel from `async-support`, a patched version of `core`, to `libtock-rs`. Have a look at the discussion in https://github.com/tock/libtock-rs/pull/103#discussion_r346731937. Maybe you know a better solution?

Reply:
Oh, I see the need now. Would you mind adding a comment that this is exported for use by async-support?

Reply:
Done!